Sources/XcodeGenKit/PathExtensions.swift

```diff
@@ -6,6 +6,18 @@ extension Path {
     public func byRemovingBase(path: Path) -> Path {
         return Path(normalize().string.replacingOccurrences(of: "\(path.normalize().string)/", with: ""))
     }
+
+    /// Returns a Path without any inner parent directory references.
+    ///
+    /// Similar to `NSString.standardizingPath`, but works with relative paths.
+    ///
+    /// ### Examples
+    /// - `a/b/../c` simplifies to `a/c`
+    /// - `../a/b` simplifies to `../a/b`
+    /// - `a/../../c` simplifies to `../c`
+    public func simplifyingParentDirectoryReferences() -> Path {
+        return normalize().components.reduce(Path(), +)
```

Comment:
Does normalize just do the same thing? What does adding the components together again do?

Reply:
normalize calls [`NSString.standardizingPath`](https://developer.apple.com/documentation/foundation/nsstring/1407194-standardizingpath#), which _almost_ does the same thing but doesn't remove parent directory references in relative paths: 

> **For absolute paths only**, resolving references to the parent directory (that is, the component “..”) to the real parent directory if possible using resolvingSymlinksInPath. For relative paths, references to the parent directory are left in place.

PathKit's + operator backs up a level if you try to add `..` to a path, which means it can handle this normalization edge case for us. In other words:

```swift
Path("foo/../bar").normalize() == Path("foo/../bar")
Path("foo") + ".." + "bar" == Path("bar")
```